lightningd/invoice.c

```diff
@@ -489,7 +497,7 @@ void invoice_try_pay(struct lightningd *ld,
 	payload->preimage = details->r;
 	payload->msat = msat;
 	payload->set = set;
-	payload->outpoint = outpoint;
+	payload->outpoint = tal_steal(payload, outpoint);
```

Comment:
Ah!  If you want your own lifetimed copy, use `tal_dup_or_null(payload, struct outpoint, outpoint)`